tests/test_clip_grads.py

```diff
@@ -37,7 +37,8 @@ def test_clip_grads_with_pp(norm_type: float):
     init_distributed(tp=1, dp=1, pp=2)(_test_clip_grads_with_pp)(norm_type=norm_type)
 
 
-def _test_clip_grads_with_pp(parallel_context: ParallelContext, norm_type: float):
+def _test_clip_grads_with_pp(tp: int, pp: int, dp: int, norm_type: float):
+    parallel_context = ParallelContext(data_parallel_size=dp, pipeline_parallel_size=pp, tensor_parallel_size=tp)
```

Comment:
this part is reused a lot let's try to refactor the parallel_context creation in mock environment